src/TorchSharp/NN/Module.cs

```diff
@@ -980,6 +1035,30 @@ protected virtual void RegisterComponents()
                     _areComponentsRegistered = true;
                 }
 
+                protected static (Device device, ScalarType dtype) GetDefaultDeviceAndType(Device device = null, ScalarType? dtype = null)
+                {
+                    if (!dtype.HasValue)
+                        dtype = get_default_dtype();
+
+                    if (device == null)
+                        device = torch.CPU;
+
+                    return (device, dtype.Value);
+                }
+
+                internal T MoveModule<T>(Device device, ScalarType? dtype) where T : Module
```

Comment:
Make this a static extension method, then C# will infer the return type.

Reply:
I'll take a look at that.